processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/FragmentCompletionTimeDeriver.java

```diff
@@ -90,4 +94,13 @@ public CompletionTime processOneToOne(String tenantId, Span item) throws RetryAt
         return null;
     }
 
+    private void setFault(CompletionTime completionTime, List<BinaryAnnotation> binaryAnnotations) {
+        List<HttpCodesUtil.HttpCode> errorCodes =
+                HttpCodesUtil.getClientOrServerErrors(HttpCodesUtil.getHttpStatusCode(binaryAnnotations));
+
+        if (errorCodes.size() > 0) {
+            completionTime.setFault(errorCodes.iterator().next().getDescription());
```

Comment:
Is there a reason to return only the first error description, instead of joining them all together?